src/clarity-angular/forms/_forms.clarity.scss

```diff
@@ -8,7 +8,7 @@
 @import "../utils/layers.clarity";
 @import "../images/icons.clarity";
 
-$styledInputs: inputs("text", "password", "number", "email", "url");
+$styledInputs: inputs("text", "password", "number", "email", "url", "tel", "date");
```

Comment:
Was this tested on all browsers? Sometimes browsers render different input types differently. For eg: for the number input type chrome and firefox show a spin box for incrementing and decrementing the numbers while IE does not (http://stackoverflow.com/questions/3790935/can-i-hide-the-html5-number-input-s-spin-box). Although we didn't disable the browser defaults for type number but we need to find out how different browsers behave and note them & check if we want to change anything.